js/apps/system/_admin/aardvark/APP/react/src/views/collections/indices/addIndex/AddIndexForm.tsx

```diff
@@ -0,0 +1,68 @@
+import { Box, Stack } from "@chakra-ui/react";
+import React, { useState } from "react";
+import SingleSelect from "../../../../components/select/SingleSelect";
+import { useCollectionIndicesContext } from "../CollectionIndicesContext";
+import { FulltextIndexForm } from "./FulltextIndexForm";
+import { GeoIndexForm } from "./GeoIndexForm";
+import { InfoTooltip } from "./InfoTooltip";
+import { PersistentIndexForm } from "./PersistentIndexForm";
+import { TTLIndexForm } from "./TTLIndexForm";
+import { ZKDIndexForm } from "./ZKDIndexForm";
+
+export const AddIndexForm = ({ onClose }: { onClose: () => void }) => {
+  const { indexTypeOptions } = useCollectionIndicesContext();
+  const [indexType, setIndexType] = useState(indexTypeOptions?.[0].value || "");
+  let tooltipText = "Type of index to create.";
+  if (!indexTypeOptions?.find(option => option.value === "hash")) {
+    tooltipText = `${tooltipText} Please note that for the RocksDB engine the index types "hash", "skiplist' and "persistent" are identical, so that they are not offered seperately here.`;
+  }
+  return (
+    <Box width="100%" padding="4">
+      <Stack padding="4" background="white" spacing="4">
+        <Box
+          display={"grid"}
+          gridTemplateColumns={"200px 1fr 40px"}
+          rowGap="5"
+          columnGap="3"
+          maxWidth="500px"
+        >
+          <Box fontSize={"lg"}>Add new index</Box>
+          <SingleSelect
+            defaultValue={indexTypeOptions?.[0]}
+            options={indexTypeOptions}
+            onChange={value => {
+              setIndexType((value as any).value);
+            }}
+          />
+          <InfoTooltip label={tooltipText} />
+        </Box>
+        <IndexTypeForm onClose={onClose} type={indexType} />
+      </Stack>
+    </Box>
+  );
+};
+
+const IndexTypeForm = ({
+  type,
+  onClose
+}: {
+  type: string;
+  onClose: () => void;
+}) => {
+  if (type === "persistent") {
```

Comment:
Would it be possible to create a proper type for `type` ? I noticed similar ones in `IndexType` . So maybe I am missing some context here. 

Example for a similar use-case
```
type IndexType = {
    fields: 'fullText' | 'primary';
}

type someType = IndexType[keyof IndexType] | 'persistent';
```

This way someType would be evaluated to.
`type someType = "fullText" | "primary" | "persistent"`

Maybe if not now, good ref for future 🙂

Reply:
fixed the types here, good catch!